theseus/labs/lie/lie_tensor.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from enum import Enum
+from typing import List, Tuple, Union
+
+import torch
+
+from theseus.labs.lie.functional.lie_group import UnaryOperatorOpFnType
+from theseus.labs.lie.functional import se3 as _se3_base
+from theseus.labs.lie.functional import so3 as _so3_base
+
+
+TensorType = Union[torch.Tensor, "LieTensor"]
+
+
+class LieType(Enum):
+    SE3 = 0
+    SO3 = 1
+
+
+SE3 = LieType.SE3
+SO3 = LieType.SO3
+
+_JFnReturnType = Tuple[List[torch.Tensor], TensorType]
+
+
+class LieTensor:
+    def __init__(self, data_tensor: torch.Tensor, ltype: LieType):
+        self._t = data_tensor
+        self._fn_lib = {
+            SE3: _se3_base,
+            SO3: _so3_base,
+        }[ltype]
+        self._fn_lib.check_group_tensor(data_tensor)
+        self.ltype = ltype
+
+    def __repr__(self) -> str:
+        return f"LieTensor({self._t}, ltype=lie.{self.ltype})"
+
+    def _check_ltype(self, other: "LieTensor", op_name: str):
+        if other.ltype != self.ltype:
+            raise ValueError("f{op_name} requires both tensors to have same ltype.")
+
+    # Returns a new LieTensor with the given data and the same ltype as self
+    def new(self, t: TensorType) -> "LieTensor":
+        tensor = t if isinstance(t, torch.Tensor) else t._t
+        return LieTensor(tensor.clone(), ltype=self.ltype)
+
+    # Operators
+    def exp(self, tangent_vector: torch.Tensor) -> "LieTensor":
+        return self.new(self._fn_lib.exp(tangent_vector))
+
+    def log(self, group: "LieTensor") -> torch.Tensor:
+        return self._fn_lib.log(group._t)
+
+    def adj(self, group: "LieTensor") -> "LieTensor":
+        return self.new(group)
```

Comment:
I know adj returns the same group but should we still call `self._fn_lib.adj` for consistency and just in case anything in the functional code changes but doesn't propagate up here.

Reply:
Oh, yeah, we shouldn't be doing this, good catch!

Reply:
Fixed. This was also flat out wrong for SE3.

Reply:
Did the tests still pass? Oh I guess we don't have tests on top of the LieTensor interface, right?

Reply:
I have added some tests now.